File: src/common/framework/camera.c
#include "GLES/gl.h"
#include "camera.h"
#include "framework/util.h"


void cameraSetActive(Camera* cam) {
  glMatrixMode(GL_PROJECTION);
  glLoadIdentity();
  glScalef(1, -1, 1);
  glOrthof(cam->posX,  cam->posX + cam->width, cam->posY, cam->posY + cam->height,-1, 1);

  glMatrixMode(GL_MODELVIEW);
}

void cameraInit(Camera* cam, float x,float y, int width, int height) {
  printf("Cam: %i, %i, %i, %i", x, y, width, height);
  cam->posX = x;
  cam->posY = y;
  cam->width = width;
  cam->height = height;
}

File: src/common/framework/camera.nim
when not defined(CAMERA_H_D4FVIPNE): 
  const 
    CAMERA_H_D4FVIPNE* = true
  type 
    Camera* {.importc: "Camera", header: "camera.h".} = object 
      posX* {.importc: "posX".}: cfloat
      posY* {.importc: "posY".}: cfloat
      width* {.importc: "width".}: cint
      height* {.importc: "height".}: cint

  proc cameraInit*(cam: ptr Camera; x: cfloat; y: cfloat; width: cint; 
                   height: cint) {.importc: "cameraInit", header: "camera.h".}
  proc cameraSetActive*(cam: ptr Camera) {.importc: "cameraSetActive", 
      header: "camera.h".}

